client/upload-kpm.py

```diff
@@ -0,0 +1,92 @@
+import time
+import os
+import paramiko
+
+import king_phisher.client.plugins as plugins
+import king_phisher.client.gui_utilities as gui_utilities
+import king_phisher.client.export as export
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Jeremy Schoeneman']
+	title = 'Upload KPM'
+	description = """
+	Saves a KPM file to the king-phisher server when sent
+	Must have keyed authentication enabled to save KPM to server
+	Also must have write permissions to the folder you're uploading to
+	"""
+	
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+
+	options = [
+		plugins.ClientOptionString(
+			'local_directory',
+			'Local directory to save KPM File',
+			default='/$HOME/',
+			display_name='Local Directory'
+		),
+		plugins.ClientOptionString(
+			'remote_directory',
+			'Directory on the Server to Upload the KPM',
+			default='/usr/share/',
+			display_name='Remote Directory'
+		),
+		plugins.ClientOptionString(
+			'key_path',
+			'Path to Pub SSH Key',
+			default='$HOME/.ssh/id_rsa.pub',
+			display_name='SSH Key Path'
+		)
+	]
+	
+	def initialize(self):
+		mailer_tab = self.application.main_tabs['mailer']
+		# self.signal_connect('send-finished', self.signal_save_kpm, gobject=mailer_tab)
+		self.signal_connect('send-precheck', self.signal_save_kpm, gobject=mailer_tab)
+		return True
+
+	def signal_save_kpm(self, config):
+		mailer_tab = self.application.main_tabs['mailer']
+		username = self.application.config['server_username']
+		current_time = time.strftime('%m-%d-%Y.%H:%M:%S')
+		campaign_name = self.application.config['campaign_name']
+		filename = username + '_' + campaign_name + '_' +  str(current_time) + '.kpm'
+		local_kpm = self.config['local_directory'] + '/' + filename
```

Comment:
use 'os.path.join(path, path, filename)' . This provides better support for both Linux and Windows user clients.